### PR TITLE
Fix requirement files and package data

### PR DESCRIPTION
**Pull request by DanielNoord**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :sparkles: New feature |
| ✓   | :hammer: Refactoring   |

## Description

Closes https://github.com/pylint-dev/astroid/issues/2129

The first commit has been a pet peeve of mine for months. I really disliked how it was hard to see what you need to install when first opening the project. I hope this lay out is a bit more explicit.

The second commit adds the files as requested by @mtelka.
Tested with `pip install .` and the files ended up in the egg info. @mtelka could you test if this branch fits your needs?

### CODE REVIEW

**Comment by DanielNoord** [comment]
Sorry for the spam guys. I'm walking to the coffee machine as we speak to hopefully prevent more commits :smile:

File: requirements_dev.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+-r requirements_minimal.txt
+
+# Tools used during development, prefer installing these with pre-commit
+black
+pre-commit
+pylint
```
Comment (Pierre-Sassoulas) [inline requirements_dev.txt:6]: pylint cannot be installed with pre-commt

File: requirements_minimal.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Tools used when releasing
+contributors-txt>=0.7.4
+tbump~=6.9.0
```
Comment (Pierre-Sassoulas) [inline requirements_minimal.txt:3]:
```suggestion
tbump~=6.9
```

File: requirements_dev.txt (hunk 1 above)
Comment (Pierre-Sassoulas) [reply requirements_dev.txt:6]: "prefer installing these with pre-commit", this is not possible for pylint as it's a system hook.

**Comment by mtelka** [comment]
I reviewed changes and they looks reasonable, but I'm not sure how to test this properly without a sdist release.  Anyway, my workflow (packaging of Python projects for OpenIndiana) does not require perfect sdists because I can easily patch them (as I currently do with astroid 2.15.4), so if there will be something still not working as expected I'll report back.

Thank you!

**Comment by hroncok** [comment]
This makes the files installed.

```
(venv) [tmp]$ pip install https://github.com/pylint-dev/astroid/archive/da246398cb42c1c0509ce572cfb93b6bc1776c1a.zip
...

(venv) [tmp]$ ls -1 venv/lib/python3.11/site-packages/
...
astroid
astroid-3.0.0a3.dev0.dist-info
...
requirements_dev.txt
requirements_full.txt
requirements_minimal.txt
...
tests
tox.ini
```

**Comment by hroncok** [comment]
Don't `package-data` it, include it in the sdist only? Back in the day, that was done using `MANIFEST.in`. No idea if it's possible to do it via `pyproject.toml`.

**Comment by DanielNoord** [comment]
@hroncok Sorry for taking so long to respond. This has been constantly on my radar, but I really have no idea how to fix this. If I `include` the files in `MANIFEST.in` and remove them from the `pyproject.toml` I still get them when doing `pip install .`. Is that expected? I'm completely lost here.

**Comment by DanielNoord** [comment]
Hmm that did seem to work. No idea why it didn't on initial try locally. Thanks for the help!

**Pull request merged**